synapse/storage/databases/main/event_push_actions.py

```diff
@@ -448,25 +452,37 @@ async def get_unread_push_actions_for_user_in_range_for_http(
             The list will be ordered by ascending stream_ordering.
             The list will have between 0~limit entries.
         """
+
         # find rooms that have a read receipt in them and return the next
         # push actions
         def get_after_receipt(
             txn: LoggingTransaction,
         ) -> List[Tuple[str, str, int, str, bool]]:
             # find rooms that have a read receipt in them and return the next
             # push actions
-            sql = """
+
+            receipt_types_clause, args = make_in_list_sql_clause(
+                self.database_engine,
+                "receipt_type",
+                (
+                    ReceiptTypes.READ,
+                    ReceiptTypes.READ_PRIVATE,
+                    ReceiptTypes.UNSTABLE_READ_PRIVATE,
+                ),
+            )
+
+            sql = f"""
                 SELECT ep.event_id, ep.room_id, ep.stream_ordering, ep.actions,
                     ep.highlight
                 FROM (
                     SELECT room_id,
                         MAX(stream_ordering) as stream_ordering
                     FROM events
                     INNER JOIN receipts_linearized USING (room_id, event_id)
-                    WHERE receipt_type = 'm.read' AND user_id = ?
+                    WHERE {receipt_types_clause} AND user_id = ?
```

Comment:
I think the ramifications of this change (and the ones below it) are that we might have HTTP pushed / emailed for something that was actually read using a private read receipt? Does that sound accurate?

Reply:
So I am not sure I understand the code at hand though my thinking was that since both public and private RRs influence notifs in the same way, they should be added here. Was my thinking incorrect?

Reply:
> Was my thinking incorrect?

That's my understanding as well. I'm trying to figure out the user-impact of fixing this bug.

Reply:
OK I double checked this; I think (before fixing this to look at all read receipt types) you could end up in this situation:

1. There are some new events in a room.
2. The user reads them and sends a private read receipt.
3. They have an email or HTTP pusher configured.
4. The private read receipt would not be considered read when checking for events to push.
5. The new events would be pushed even though they're before the private read receipt.

Good that we found it, but doesn't seem to be the end of the world!

Reply:
> Good that we found it

Thank you very much for doing that! I've totally missed it